FreeRTOS/FreeRTOS-Plus-TCP
# Fix unit tests wrt. latest dev/IPv6_Integration changes

Description
-----------
This PR fixes the unit test failures for all unit tests modules which were failing after the latest IPv6 integration source changes. The `xEndPoint` structures that are used in the test cases are now initialized to zero, to prevent uninitialized data from previous tests affecting the behavior of the tests.

The CBMC proof for `DNSgetHostByName_a` is updated to not assert on `pxAddressInfo != NULL` during the DNS callback as `pxAddressInfo` being NULL is a valid case during timeouts. 

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ 
make -C test/unit-test/build/ all 
cd test/unit-test/build/
ctest -E system --output-on-failure --verbose
```

Checklist:
----------

Related Issue
-----------
Unit test cases were failing due to the changes in the source code as part of the IPv6 integration.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
